test/libsolidity/syntaxTests/nameAndTypeResolution/363_non_payable_constructor.sol

```diff
@@ -3,9 +3,9 @@ contract C {
 }
 contract D {
     function f() public returns (uint) {
-        (new C).value(2)();
+        (new C){value: 2}();
         return 2;
     }
 }
 // ----
-// TypeError: (106-119): Constructor for contract C must be payable for member "value" to be available.
+// TypeError: (106-123): Cannot set option "value" on a non-payable function type.
```

Comment:
This actually hints at a separate issue - the error message should be the same with the new syntax as it was for the old syntax. We could either fix this in this PR or create an issue for it and fix that separately.

Reply:
I agree with you. I can fix message to be the same like before as part of this PR.